jbrowse/src/org/labkey/jbrowse/JBrowseController.java

```diff
@@ -962,14 +962,18 @@ public ApiResponse execute(LuceneQueryForm form, BindException errors)
 
             try
             {
-                return new ApiSimpleResponse(searcher.doSearchJSON(
+                HttpServletResponse response = getViewContext().getResponse();
+                response.setContentType("application/x-ndjson");
```

Comment:
what's ndjson?

Reply:
ndjson is newline-delimited JSON. Instead of returning a JSON array, the data is a set of newline-delimited JSON objects, which means a JSON parser can parse them individually instead of waiting for the array to close. It's technically not valid JSON, so it requires a separate content type header, but it allows us to stream and parse incrementally on the client side instead of waiting for the whole query to return.